apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx

```diff
@@ -163,6 +163,23 @@ const SwapForm = () => {
     // However, if you wish to use a different fee type, you will need to set it as custom.
     const isCustomFee =
       feesStrategy === "slow" || feesStrategy === "fast" || feesStrategy === "custom";
+
+    const providerRedirectFromAccountId =
+      fromAccount &&
+      provider &&
+      walletApiPartnerList?.enabled &&
+      walletApiPartnerList?.params?.list.includes(provider)
+        ? accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id
+        : fromAccount?.id;
```

Comment:
```suggestion
    const providerRedirectFromAccountId = accountToWalletAPIAccount(fromAccount, fromParentAccount)?.id;
```

Are we going to be always in wallet-api context? Do we really need this if/else?
(Inside getSwapWebAppInputProps, behind isSwapLiveAppEnabled)

Reply:
You are totally wright, I have another thing in mind.